packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol

```diff
@@ -92,26 +92,29 @@ contract OptimisticAssertor is Lockable, OptimisticAssertorInterface, Ownable {
             callbackRecipient: callbackRecipient,
             sovereignSecurityManager: sovereignSecurityManager,
             currency: currency,
-            useDisputeResolution: true, // this is the default behavior: if not specified by the Sovereign security manager the assertion will respect the DVM result.
-            useDvmAsOracle: true, // this is the default behavior: if not specified by the Sovereign security manager the assertion will use the DVM as an oracle.
             settled: false,
             settlementResolution: false,
             bond: bond,
             assertionTime: getCurrentTime(),
-            expirationTime: getCurrentTime() + liveness
+            expirationTime: getCurrentTime() + liveness,
+            claimId: keccak256(claim),
+            ssmSettings: SsmSettings({
+                useDisputeResolution: true, // this is the default behavior: if not specified by the Sovereign security manager the assertion will respect the DVM result.
+                useDvmAsOracle: true // this is the default behavior: if not specified by the Sovereign security manager the assertion will use the DVM as an oracle.
```

Comment:
Had to use nested properties to avoid stack too deep error

Reply:
I think this is unavoidable. in the related PR https://github.com/UMAprotocol/protocol/pull/4254 I modified this struct to store more ssm related settings.